glide-core/src/client/value_conversion.rs

```diff
@@ -607,6 +608,50 @@ pub(crate) fn convert_to_expected_type(
             }
             _ => Err((ErrorKind::TypeError, "Response couldn't be converted").into()),
         },
+        // Used by XAUTOCLAIM. The command returns a list of length 2 if the Redis version is less than 7.0.0 or a list
+        // of length 3 otherwise. It has the following response format:
+        /* Redis version < 7.0.0 example:
+        1) "0-0"
+        2) 1) 1) "1-0"
+              2) 1) "field1"
+                 2) "value1"
+           2) 1) "1-1"
+              2) 1) "field2"
+                 2) "value2"
+
+        Redis version >= 7.0.0 example:
+        1) "0-0"
+        2) 1) 1) "1-0"
+              2) 1) "field1"
+                 2) "value1"
+           2) 1) "1-1"
+              2) 1) "field2"
+                 2) "value2"
+        3) 1) "1-1"
+        */
+        ExpectedReturnType::XAutoClaimReturnType => match value {
+            // Response will have 2 elements if Redis version < 7.0.0, and 3 elements otherwise.
+            Value::Array(ref array) if array.len() == 2 || array.len() == 3 => {
+                let mut result: Vec<Value> = Vec::with_capacity(array.len());
+                result.push(array[0].clone());
+                // Only the element at index 1 needs conversion.
+                result.push(convert_to_expected_type(array[1].clone(), Some(ExpectedReturnType::Map {
```

Comment:
are there performance concerns with cloning array[1] (a map of keys to their field/value pairs) or array[2] (a list of stream IDs that were in the pending entries list but have been deleted from the stream)? If so, what would the alternative way to do this be?

Reply:
You can do `array.remove(N)` which pops elements from the array (note: indices get shifted). See processing of `FunctionStatsReturnType` for example`.

Reply:
If you don't want the indices to shift, https://doc.rust-lang.org/std/vec/struct.Vec.html#method.swap_remove is a bit more performant.